fakestorage/server.go

```diff
@@ -287,6 +291,85 @@ func (s *Server) buildMuxer() {
 	s.mux.Host("{bucketName:.+}").Path("/{objectName:.+}").Methods(http.MethodPost, http.MethodPut).HandlerFunc(jsonToHTTPHandler(s.insertObject))
 }
 
+func (s *Server) reseedServer(r *http.Request) jsonResponse {
+	folder := "/data"
+	initialObjects, emptyBuckets := generateObjectsFromFiles(folder)
+	for _, bucketName := range emptyBuckets {
+		s.CreateBucketWithOpts(CreateBucketOpts{Name: bucketName})
+	}
```

Comment:
hmm this is tricky because if we reassign the backend below, those buckets are going to be gone. Perhaps the backend constructor could take a list of empty buckets too?

Reply:
That is a good point. However, when I run the following script, where three empty buckets are created before the reseed, it seems like those buckets are detected:
```
docker run -d --name fake-gcs-server -p 4443:4443 -v ${PWD}/all_files:/data fake-server-reseed
mkdir ./all_files/bucket2
mkdir ./all_files/bucket3
mkdir ./all_files/bucket4

echo "Before Reseed:"
for i in {2..4}; do
    echo "Contents of bucket$i"
    curl -k "https://localhost:4443/storage/v1/b/bucket$i/o"
done

echo "Hitting the reseed endpoint:"
curl -k -X POST "https://localhost:4443/_internal/reseed"
echo ""

echo "After Reseed:"
for i in {2..4}
do
    echo "Contents of bucket$i"
    curl -k "https://localhost:4443/storage/v1/b/bucket$i/o"
done
```

Output:
```
...
After Reseed:
Contents of bucket2
{"kind":"storage#objects","items":[]}
Contents of bucket3
{"kind":"storage#objects","items":[]}
Contents of bucket4
{"kind":"storage#objects","items":[]}
```

However, I definitely see the merit of passing in the `emptyBuckets` into the backend `NewStorageFS` and `NewStorageMemory` functions as that would also simplify `main.go`. I can push a change for this.